applications/image_demo/image_pipeline.cpp

```
@@ -0,0 +1,230 @@
+
+	auto noisy = add_noise(scene, 0.08, 42);
+	print_stats(noisy, "noisy scene");
+	io::write_pgm("02_noisy.pgm", noisy);
+	std::cout << "  -> 02_noisy.pgm\n";
+
+	// ================================================================
+	// Stage 3: Denoise with Gaussian blur
+	// ================================================================
+	print_separator("Stage 3: Gaussian Blur (sigma=1.5)");
+
+	auto denoised = gaussian_blur(noisy, 1.5);
```

Cppcheck (2.20.0)

[error] Unhandled exception thrown in function that is an entry point.

(throwInEntryPoint)